internal/guest/runtime/hcsv2/spec.go

```diff
@@ -152,3 +167,68 @@ func getGroup(spec *oci.Spec, filter func(user.Group) bool) (user.Group, error)
 	}
 	return groups[0], nil
 }
+
+// applyAnnotationsToSpec modifies the spec based on additional information from annotations
+func applyAnnotationsToSpec(ctx context.Context, spec *oci.Spec) error {
+	// Check if we need to override container's /dev/shm
+	if val, ok := spec.Annotations["io.microsoft.container.storage.shm.size-kb"]; ok {
+		sz, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return errors.Wrap(err, "/dev/shm size must be a valid integer")
+		}
+		if sz <= 0 {
+			return errors.Errorf("/dev/shm size must be a positive integer, got: %d", sz)
+		}
+
+		// Use the same options as in upstream https://github.com/containerd/containerd/blob/main/oci/mounts.go#L49
```

Comment:
This links directly to the main branch right? If someone changed this file I think the line number will be off then.